File: android/app/src/main/java/com/kufarcode/app/data/remote/dto/CityDto.kt
package com.kufarcode.app.data.remote.dto

data class CityDto(
    val code: String,
    val name: String,
    val theme: CityThemeDto,
    val features: CityFeatureFlagsDto
)

data class CityThemeDto(
    val primaryColor: String,
    val secondaryColor: String?,
    val logoUrl: String?
)

data class CityFeatureFlagsDto(
    val enableMap: Boolean?,
    val enableFavorites: Boolean?,
    val experimentalLayout: Boolean?
)

File: KufarCodeApp/Data/DTO/CityDto.swift
import Foundation

struct CityDto: Codable {
    let code: String
    let name: String
    let theme: CityThemeDto?
}

struct CityThemeDto: Codable {
    let primaryColor: String
}

extension CityDto {
    func toDomain() -> City {
        City(
            code: code,
            name: name,
            theme: theme?.toDomain()
        )
    }
}

extension CityThemeDto {
    func toDomain() -> CityTheme {
        CityTheme(primaryColorHex: primaryColor)
    }
}
